drivers/CoreRFIDReader/tests/CoreRFIDReader_test.cpp

```diff
@@ -0,0 +1,264 @@
+// Leka - LekaOS
+// Copyright 2021 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+#include <functional>
+
+#include "CoreRFIDReader.h"
+#include "gtest/gtest.h"
+#include "mocks/leka/CoreBufferedSerial.h"
+#include "mocks/leka/EventQueue.h"
+
+using namespace leka;
+using namespace interface;
+
+using ::testing::Args;
+using ::testing::DoAll;
+using ::testing::ElementsAre;
+using ::testing::InSequence;
+using ::testing::Return;
+using ::testing::SaveArg;
+using ::testing::SetArrayArgument;
+
+class CoreRFIDReaderTest : public ::testing::Test
+{
+  protected:
+	CoreRFIDReaderTest() : reader(mockBufferedSerial) {};
+
+	// void SetUp() override {}
+	// void TearDown() override {}
+
+	mock::EventQueue event_queue {};
+	mock::CoreBufferedSerial mockBufferedSerial;
+	CoreRFIDReader reader;
+
+	bool _callbackCalled {false};
+
+	static void func1() {};
+	static void func2(rfid::Tag tag) {};
+	std::function<void()> callbackDetected			= func1;
+	std::function<void(rfid::Tag)> callbackReadable = func2;
+	std::function<void()> callbackSigio {};
```

Comment:
Transforme tes callbacks callbackDetected et callbackReadable en MockFunction, tu verras qu'il y a des appels en plus et ça permettra d'avoir plus de contrôle sur ce qui se passe

```suggestion
	testing::MockFunction<void()> callback_detected;
	testing::MockFunction<void(rfid::Tag)> callback_readable;
	std::function<void()> callbackSigio {};
```

Reply:
Rappel: lors du register, il ne faut pas passer la callback mais appeler `AsStdFunction()`

Exemple:  
`reader.registerOnTagDetectedCallback(callbackDetected.AsStdFunction());`